### ISSUE TITLE
NoWhitespaceAfter synchronized method

### ISSUE THREAD

**Issue by ctabin** [checkstyle/checkstyle#12486]
Hello,

I have the following checkstyle configuration in `pom.xml`:
```xml
<module name="NoWhitespaceAfter">
      <property name="tokens" value="ARRAY_INIT,AT,INC,DEC,UNARY_MINUS,UNARY_PLUS,BNOT,LNOT,DOT,TYPECAST,ARRAY_DECLARATOR,INDEX_OP,LITERAL_SYNCHRONIZED,METHOD_REF" />
</module>
```

With a `mvn verify`, the following piece of code reports an error `'synchronized' is followed by whitespace`:

```java
public class TestClass {
    private synchronized Row waitForNextRow() {
    }
}
```

It seems to be a false positive because the check should only apply on structures like:
```java
synchronized(lock) { ... }
```

Here is the output:
```
$ cat TestClass.java
public class TestClass {
    private synchronized Row waitForNextRow() {
    }
}

$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
<module name="NoWhitespaceAfter">
      <property name="tokens" value="ARRAY_INIT,AT,INC,DEC,UNARY_MINUS,UNARY_PLUS,BNOT,LNOT,DOT,TYPECAST,ARRAY_DECLARATOR,INDEX_OP,LITERAL_SYNCHRONIZED,METHOD_REF" />
</module>
    </module>
</module>

$ java -jar checkstyle-10.5.0-all.jar -c TestConfig.xml TestClass.java
Starting audit...
[ERROR] TestClass.java:2:13: 'synchronized' is followed by whitespace. [NoWhitespaceAfter]
Audit done.
Checkstyle ends with 1 errors.
```

Thanks & best regards.

**Comment by ctabin** [comment]
@rnveach Oops it seems you edited my comments and I edited them back behind, sorry. Don't hesitate if you think I'm missing some important information.

**Comment by ctabin** [comment]
@rnveach Thanks. I put back the output you've added previously. Hope it is good.

**Comment by nrmancuso** [comment]
I am on it
